### PR TITLE
feat(Manager): Enhanced activity stream messages, developed custom hooks and a virtual path module, and reorganized the Symlink section.

### PR DESCRIPTION
**Pull request by nikhitamore**
… web properties table.

## Explain the feature/fix

Certainly! Here's a breakdown of the changes made in the pull request:

1. **Added 'created by' field in activity stream messages**:
   - Implemented the addition of a 'created by' field to activity stream messages.
   - Updated AS for created and updated message for virtual path.

2. **Enhanced custom hooks**:
   - Developed custom hooks to facilitate data retrieval and deletion of virtual paths under the spa property queries.
   - Introduced type definitions for improved clarity and reliability.

3. **Created virtual path module**:
   - Established a new module responsible for managing virtual paths.
   - This module includes:
     - Addition of virtual paths.
     - Deletion of existing virtual path configurations.
     - Display functionality for virtual paths.

4. **Updated Symlink section**:
   - Integrated a card component for the Symlink section.
   - Moved the Symlink functionality into a dedicated 'components' folder for better organization.

## Screenshot(s)

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [ ] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [ ] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/DeleteVirtualPath.tsx [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+          <Controller
+            control={control}
+            name="virtualPath"
+            defaultValue={data.virtualPath}
+            render={({ field, fieldState: { error } }) => (
+              <FormGroup
+                label={
+                  <>
+                    Virtual Path
+                    <Tooltip content="Symlink basePath path should be /{spa-path}/{basePath-path}">
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/DeleteVirtualPath.tsx:201]: Please remove the symlink from here, change it to the Virtual Path reference

File: packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    .string()
+    .label('Source File Path')
+    .trim()
+    .required()
+    .test(
+      'no-leading-or-trailing-slashes',
+      'Source File Path should not start or end with a slash',
+      (value) => {
+        if (value) {
+          return !/^\/+|\/+$/.test(value);
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx:36]: Wrap this regex in to variable

Comment (soumyadip007) [reply packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx:36]: `const testValue = !/^\/+|\/+$/;`

File: packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+    .string()
+    .label('Target File Path')
+    .trim()
+    .required()
+    .test(
+      'no-leading-or-trailing-slashes',
+      'Target File Path should not start or end with a slash',
+      (value) => {
+        if (value) {
+          return !/^\/+|\/+$/.test(value);
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx:51]: reuse the wrapped variable here
`const testValue = !/^\/+|\/+$/;`

File: packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+              isRequired
+              helperTextInvalid={error?.message}
+            >
+              <FormSelect
+                label="Select Environment"
+                aria-label="FormSelect Input"
+                onChange={onChange}
+                value={value}
+              >
+                <FormSelectOption key="default" label="Please select an environment" isDisabled />
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx:245]: Is it required here ! If not needed we may remove this

File: packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/DeleteVirtualPath.tsx [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+            name="basePath"
+            defaultValue={data?.basePath}
+            render={({ field, fieldState: { error } }) => (
+              <FormGroup
+                label={
+                  <>
+                    Base Path
+                    <Tooltip content="Symlink virtualPath path should be /{spa-path}/{virtualPath-path}">
+                      <span>
+                        &nbsp; <InfoCircleIcon style={{ color: '#6A6E73' }} />
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/DeleteVirtualPath.tsx:167]: Remove this tooltip, as it's not related to the virtual path, or modify accordingly

File: packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+  propertyIdentifier: yup.string().label('Web property').trim(),
+  identifier: yup.string().label('Application Name').trim(),
+  virtualPath: yup
+    .string()
+    .label('Source File Path')
+    .trim()
+    .required()
+    .test(
+      'no-leading-or-trailing-slashes',
+      'Source File Path should not start or end with a slash',
```
Comment (soumyadip007) [inline packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx:33]: Please remove this

**Review by soumyadip007 — CHANGES_REQUESTED** [review]
@nikhitamore please make the required changes

File: packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/DeleteVirtualPath.tsx (hunk 1 above)
Comment (nikhitamore) [reply packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/DeleteVirtualPath.tsx:201]: done

Comment (nikhitamore) [reply packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/DeleteVirtualPath.tsx:201]: done

File: packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx (hunk 2 above)
Comment (nikhitamore) [reply packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx:36]: done

File: packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/DeleteVirtualPath.tsx (hunk 5 above)
Comment (nikhitamore) [reply packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/DeleteVirtualPath.tsx:167]: done

File: packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx (hunk 6 above)
Comment (nikhitamore) [reply packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx:33]: We need this right for the path?

**Review by soumyadip007 — APPROVED** [review]
LGTM

File: packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx (hunk 6 above)
Comment (soumyadip007) [reply packages/manager/src/views/SPAPropertyDetailPage/VirtualPath/components/CreateVirtualPath.tsx:34]: You can change it to base path or virtual path, as it's not the file path

**Pull request merged**